festim/hydrogen_transport_problem.py

```diff
@@ -480,10 +481,52 @@ def create_dirichletbc_form(self, bc):
 
         return form
 
+    def create_source_values_fenics(self):
+        """For each source in the model, check if the species and volumes are lists,
+        create one if not, check if each element in species and volumes are
+        festim.Species and festim.VolumeSubdomain1D, then create the value fenics
+        attribute.
+        """
+        for source in self.sources:
+            # iterate through species, find species if name of species is given
+            if not isinstance(source.species, list):
+                source.species = [source.species]
+            for idx, spe in enumerate(source.species):
+                if isinstance(spe, str):
+                    # if name of species is given then replace with species object
+                    source.species[idx] = F.find_species_from_name(spe, self.species)
+
+            # iterate through volumes, find volume if id of volume is given
+            if not isinstance(source.volume, list):
+                source.volume = [source.volume]
+            for idx, vol in enumerate(source.volume):
+                if isinstance(vol, int):
+                    # if name of species is given then replace with species object
+                    source.volume[idx] = F.find_volume_from_id(
+                        source.volume, self.volume_subdomains
```

Comment:
source.volume is a list here, so I don't expect this to work,

It would rather be:

```suggestion
                    source.volume[idx] = F.find_volume_from_id(
                        source.volume[idx], self.volume_subdomains
```